erpnext/selling/doctype/sales_order/sales_order.js

```diff
@@ -35,6 +35,25 @@ frappe.ui.form.on("Sales Order", {
 			};
 		})
 	},
+	customer: function (frm) {
+		frappe.call({
+			method: "erpnext.stock.doctype.delivery_trip.delivery_trip.get_delivery_window",
+			args: { customer: frm.doc.customer },
+			callback: function (r) {
+				if (r.message && (r.message.delivery_start_time || r.message.delivery_end_time)) {
+					frappe.model.set_value(frm.doc.doctype, frm.docname, "delivery_start_time", r.message.delivery_start_time);
```

Comment:
You can just use `frm.set_value` instead to use context (this will only work for parent fields, for child tables, `frappe.model.set_value` is fine).

```suggestion
					frm.set_value("delivery_start_time", r.message.delivery_start_time);
```